src/js/app.js

```diff
@@ -0,0 +1,65 @@
+$('document').ready(function() {
+
+  //findPics wittles down an object to an array of desired data
+  //by using .filter() to filter out objects that don't meet a certain requirement
+  //and by using .map()  to transform the filtered object in to an array of desired data
+  var findPics = function(dataObj) {
+    var filterResult = dataObj.data.children.filter(function(element) {
+      if (element.data.hasOwnProperty('preview')) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+
+    var urlArray = filterResult.map(function(element) {
+      return element.data.preview.images[0].source.url;
+    });
+
+    var index = 0;
+    intervalID = window.setInterval(function() {
```

Comment:
Note that when `var` isn't used, the declaration is "hoisted" to the top. It's recommended to declare this using `var` in global scope so it's more clear what you're doing.